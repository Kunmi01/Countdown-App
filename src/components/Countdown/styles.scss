@import '../../scss/base.scss';

$UNIT_VALUE_SIZE_DESKTOP: 120px;
$SEPARATOR_SIZE_DESKTOP: 100px;

.countdown {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;

  &__unit {
    text-align: center;
    h1 {
      font-family: $font-play-bold;
      font-size: $font-xxxl;

      @include bp-from('mobile') {
        font-size: $UNIT_VALUE_SIZE_DESKTOP;
      }
    }
  }

  &__separator {
    font-size: $font-xxl;

    @include bp-from('mobile') {
      font-size: $SEPARATOR_SIZE_DESKTOP;
    }
  }
}
