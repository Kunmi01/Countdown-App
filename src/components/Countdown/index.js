import React from 'react';
import PropTypes from 'prop-types';
import convertMillisecondsToDurationObject from '../../utils/millisecondsConverter';
import './styles.scss';

const Countdown = ({ duration }) => {
  const { hours, minutes, seconds } = convertMillisecondsToDurationObject(
    duration
  );

  return (
    <div className="countdown">
      <div className="countdown__unit countdown__unit--hours">
        <h1>{hours}</h1>
        <p>Hours</p>
      </div>
      <h1 className="countdown__separator">:</h1>
      <div className="countdown__unit countdown__unit--minutes">
        <h1>{minutes}</h1>
        <p>Minutes</p>
      </div>
      <h1 className="countdown__separator">:</h1>
      <div className="countdown__unit countdown__unit--seconds">
        <h1>{seconds}</h1>
        <p>Seconds</p>
      </div>
    </div>
  );
};

Countdown.propTypes = {
  duration: PropTypes.number
};

Countdown.defaultProps = {
  duration: 0
};

export default Countdown;
