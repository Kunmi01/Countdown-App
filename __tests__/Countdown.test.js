import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Countdown from '../src/components/Countdown';

describe('Countdown', () => {
  const wrapper = mount(<Countdown duration={43685000} />);

  it('renders Countdown component', () => {
    expect(wrapper).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('displays duration in correct format', () => {
    const hourContainer = wrapper.find('.countdown__unit--hours h1');
    const minuteContainer = wrapper.find('.countdown__unit--minutes h1');
    const secondContainer = wrapper.find('.countdown__unit--seconds h1');

    expect(hourContainer.text()).toEqual('12');
    expect(minuteContainer.text()).toEqual('08');
    expect(secondContainer.text()).toEqual('05');
  });
});
