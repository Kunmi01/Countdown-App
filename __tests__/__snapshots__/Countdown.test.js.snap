// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countdown renders Countdown component 1`] = `
<Countdown
  duration={43685000}
>
  <div
    className="countdown"
  >
    <div
      className="countdown__unit countdown__unit--hours"
    >
      <h1>
        12
      </h1>
      <p>
        Hours
      </p>
    </div>
    <h1
      className="countdown__separator"
    >
      :
    </h1>
    <div
      className="countdown__unit countdown__unit--minutes"
    >
      <h1>
        08
      </h1>
      <p>
        Minutes
      </p>
    </div>
    <h1
      className="countdown__separator"
    >
      :
    </h1>
    <div
      className="countdown__unit countdown__unit--seconds"
    >
      <h1>
        05
      </h1>
      <p>
        Seconds
      </p>
    </div>
  </div>
</Countdown>
`;
